Sources/SMHS (iOS)/Views/Banners/BannerExtensions.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  BottomBlur.swift
+//  SMHS (iOS)
+//
+//  Created by Lampeh on 7.08.2022.
+//
+
+import SwiftUI
+import Kingfisher
+
+struct BannerImage: View {
+    let url: URL
+    
+    var body: some View {
+        KFImage(url)
+            .placeholder {
+                Color(UIColor.systemGray)
+            }
+            .retry(maxCount: 3, interval: .seconds(3))
+            .onFailure {
+                #if DEBUG
+                    print("failure: \($0)")
+                #endif
+                }
+            .resizable()
+            .shadow(x: 0, y: -10, blur: 15)
+            .scaledToFit()
+    }
+}
+
+extension Text {
+    func bannerTitle() -> Text {
+        self
+            .font(.title2)
+            .fontWeight(.bold)
+            .foregroundColor(.white)
+    }
+    func bannerHeadline() -> Text {
+        self
+            .font(.body)
+            .foregroundColor(.gray)
+            .fontWeight(.semibold)
+    }
+    
+    func bannerFootnote() -> Text {
+        self
+            .foregroundColor(.white)
+    }
+}
+
+extension KFImage {
+    func bottomBlurred(level: Int) -> some View {
+        return self
+            .overlay {
+                ForEach(0..<level) { _ in
```

Comment:
Since it's not working with KFImage, maybe instead of a ForEach, try just using a `.blur()` and a mask, with LinearGradient `Stops`, where location can be set from 0 to 1 to control the amount of blur.